inference-engine/samples/benchmark_app/remote_blobs_filling.cpp

```diff
@@ -0,0 +1,139 @@
+        inputBufferData[i] = static_cast<T>(distribution(gen));
+    }
+}
+
+void fillBuffer(void* inputBuffer, size_t elementsNum, Precision precision) {
+    if (precision == InferenceEngine::Precision::FP32) {
+        fillBufferRandom<float, float>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::FP16) {
+        fillBufferRandom<short, short>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::I32) {
+        fillBufferRandom<int32_t, int32_t>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::I64) {
+        fillBufferRandom<int64_t, int64_t>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::U8) {
+        // uniform_int_distribution<uint8_t> is not allowed in the C++17
+        // standard and vs2017/19
+        fillBufferRandom<uint8_t, uint32_t>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::I8) {
+        // uniform_int_distribution<int8_t> is not allowed in the C++17 standard
+        // and vs2017/19
+        fillBufferRandom<int8_t, int32_t>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::U16) {
+        fillBufferRandom<uint16_t, uint16_t>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::I16) {
+        fillBufferRandom<int16_t, int16_t>(inputBuffer, elementsNum);
+    } else if (precision == InferenceEngine::Precision::BOOL) {
+        fillBufferRandom<uint8_t, uint32_t>(inputBuffer, elementsNum, 0, 1);
+    } else {
+        IE_THROW() << "Requested precision is not supported";
+    }
+}
+
+size_t getBytesPerElement(Precision precision) {
+    switch (precision) {
+    case InferenceEngine::Precision::FP32:
+        return 4;
+    case InferenceEngine::Precision::FP16:
+        return 2;
+    case InferenceEngine::Precision::I32:
+        return 4;
+    case InferenceEngine::Precision::I64:
+        return 8;
+    case InferenceEngine::Precision::U8:
+        return 1;
+    case InferenceEngine::Precision::I8:
+        return 1;
+    case InferenceEngine::Precision::U16:
+        return 2;
+    case InferenceEngine::Precision::I16:
+        return 2;
+    case InferenceEngine::Precision::BOOL:
+        return 1;
+    default:
+        IE_THROW() << "Requested precision is not supported";
+    }
+}
+
+void fillRemoteBlobs(const std::vector<std::string>& inputFiles,
+                     const size_t& batchSize,
+                     benchmark_app::InputsInfo& app_inputs_info,
+                     std::vector<InferReqWrap::Ptr> requests,
+                     const ExecutableNetwork& exeNetwork) {
+#if defined(OPENCL_FOUND)
+    slog::info << "Device memory will be used for input and output blobs" << slog::endl;
+    if (inputFiles.size()) {
+        slog::warn << "Device memory supports only random data at this momnent, input images will be ignored"
```

Comment:
`momnent` -> `moment`

Reply:
If random is only supported, we won't be able to measure performance with NMS or DO. Is it acceptable?

Reply:
Thank you, fixed
As for adding input images support, I think it is acceptable for an initial implementation, but we may consider extending support for different input types in the future